vault/resource_managed_keys.go

```diff
@@ -0,0 +1,473 @@
+		},
+		"slot": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "The slot number to use, specified as a string in a " +
+				"decimal format (e.g. '2305843009213693953')",
+		},
+		"token_label": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The PIN for login",
+		},
+		"curve": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "Supplies the curve value when using " +
+				"the 'CKM_ECDSA' mechanism. Required if " +
+				"'allow_generate_key' is true",
+		},
+		"key_bits": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "Supplies the size in bits of the key when using " +
+				"'CKM_RSA_PKCS_PSS', 'CKM_RSA_PKCS_OAEP' or 'CKM_RSA_PKCS' " +
+				"as a value for 'mechanism'. Required if " +
+				"'allow_generate_key' is true",
+		},
+		"force_rw_session": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The PIN for login",
+		},
+	}
+
+	return setCommonManagedKeysSchema(s)
+}
+
+func managedKeysAWSConfigSchema() schemaMap {
+	s := schemaMap{
+		consts.FieldName: {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+			Description: "A unique lowercase name that serves as " +
+				"identifying the key",
+		},
+		"access_key": {
+			Type:     schema.TypeString,
+			Required: true,
+			Description: "The AWS access key to use. This can also " +
+				"be provided with the 'AWS_ACCESS_KEY_ID' env variable",
+		},
+		"secret_key": {
+			Type:     schema.TypeString,
+			Required: true,
+			Description: "The AWS secret key to use. This can also " +
+				"be provided with the 'AWS_SECRET_ACCESS_KEY' env variable",
+		},
+		"curve": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "The curve to use for an ECDSA key. Used " +
+				"when key_type is 'ECDSA'. Required if " +
+				"'allow_generate_key' is true",
+		},
+		"endpoint": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Used to specify a custom AWS endpoint",
+		},
+		"key_bits": {
+			Type:     schema.TypeString,
+			Required: true,
+			Description: "The size in bits for an RSA key. This " +
+				"field is required when 'key_type' is 'RSA'",
+		},
+		"key_type": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The type of key to use",
+		},
+		"kms_key": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "An identifier for the key",
+		},
+		"region": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "us-east-1",
+			Description: "The AWS region where the keys are stored (or will be stored)",
+		},
+	}
+
+	return setCommonManagedKeysSchema(s)
+}
+
+func managedKeysAzureConfigSchema() schemaMap {
+	s := schemaMap{
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+			Description: "A unique lowercase name that serves as " +
+				"identifying the key",
+		},
+		"tenant_id": {
```

Comment:
Since this is new code, I would recommend pulling all of these out to constants, adding them to the `consts` package should be fine.

Reply:
Agreed. Factored out the fields into the `internal/consts` package